### PR TITLE
chore: Improve CompareTrials perfomance

### PR DESCRIPTION
**Pull request by AmanuelAaron**
## Ticket

ET-685

## Description

Improving performance of run comparison.

## Test Plan

Select a large amount of runs (>50) on the Run Table and click compare. Ensure that the time it takes to load is resonalble

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/db/postgres_experiments.go [hunk 1]
```diff
@@ -521,6 +521,24 @@ WHERE t.id = ?`, trialID).Scan(ctx, &experiment); err != nil {
 	return &experiment, nil
 }
 
+// ExperimentsByTrialID looks up an experiment by a given list of trialIDs, returning
+// an error if none exists.
+func ExperimentsByTrialID(ctx context.Context, trialIDs []int) ([]*model.Experiment, error) {
```
Comment (carolinaecalderon) [inline master/internal/db/postgres_experiments.go:526]: blocker -- missing a test?

File: master/internal/trials/utils.go [hunk 2]
```diff
@@ -42,3 +42,32 @@ func CanGetTrialsExperimentAndCheckCanDoAction(ctx context.Context,
 	}
 	return nil
 }
+
+// CanGetTrialsExperimentAndCheckCanDoActionBulk functions the same as
+// CanGetTrialsExperimentAndCheckCanDoAction but takes in multiple trial ids.
+func CanGetTrialsExperimentAndCheckCanDoActionBulk(ctx context.Context,
```
Comment (carolinaecalderon) [inline master/internal/trials/utils.go:48]: test?

File: master/internal/api_trials_intg_test.go [hunk 3]
```diff
@@ -712,12 +712,12 @@ func TestTrialAuthZ(t *testing.T) {
 			})
 			return err
 		}, true},
-		{"CanGetExperimentArtifacts", func(id int) error {
-			_, err := api.CompareTrials(ctx, &apiv1.CompareTrialsRequest{
-				TrialIds: []int32{int32(id)},
-			})
-			return err
-		}, false},
+		// {"CanGetExperimentArtifacts", func(id int) error {
```
Comment (carolinaecalderon) [inline master/internal/api_trials_intg_test.go:715]: is this an intended change? if not, please uncomment

**Review by carolinaecalderon — COMMENTED** [review]
looks good, missing some test coverage

File: master/internal/api_trials_intg_test.go (hunk 3 above)
Comment (AmanuelAaron) [reply master/internal/api_trials_intg_test.go]: Sorry there was an error with this that I forgot to fix. Done now.

File: master/internal/db/postgres_experiments.go (hunk 1 above)
Comment (AmanuelAaron) [reply master/internal/db/postgres_experiments.go:526]: Test added.

**Review by mackrorysd — APPROVED** [review]
Thanks for adding this to the performance tests - the results look good!

File: master/internal/trials/utils.go (hunk 2 above)
Comment (carolinaecalderon) [reply master/internal/trials/utils.go:50]: +1  to Sean's point.
Also `CanGetTrialsExperimentAndCheckCanDoAction` is over a year old at this point, and, if this is correctly categorized as a chore, then I think we should spend a little more time and improve the test coverage in this same ticket

Comment (AmanuelAaron) [reply master/internal/trials/utils.go:44]: Added tests. Had to move the GetUser call outside of the function. Updated the functions that call `CanGetTrialsExperimentAndCheckCanDoAction ` as well.

**Review by carolinaecalderon — APPROVED** [review]
LGTM

**Pull request merged**